core/src/main/java/org/springframework/security/access/expression/method/DefaultMethodSecurityExpressionHandler.java

```diff
@@ -151,9 +153,8 @@ private <T> Object filterCollection(Collection<T> filterTarget, Expression filte
 			}
 		}
 		this.logger.debug(LogMessage.format("Retaining elements: %s", retain));
-		filterTarget.clear();
```

Comment:
For passivity reasons, instead of changing the default strategy, what if the code first tried to modify the existing collection, and then if that fails, return the new instance? In this way, applications that are dependent on a specific concrete collection type will still continue to function.